### PR TITLE
Wip solution

### PR DESCRIPTION
**Pull request by azat-co**
CC @AnnJohn @vicfriedman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,43 @@
...
+## Objectives
+
+1. Implement a JSON config file 
+1. Implement file which imports JSON
+1. Use JSON in the file/script
+
+## Introduction
+
+In this lab, you'll learn how to import JSON files. This can be useful for data (e.g., list of customers) or configuration (e.g., server settings). In other words, JSON files are good for storing static information, i.e., information without JavaScript/Node code.
+
```
Comment (victhevenot) [inline README.md:14]: What specifically are the going to be working on now? This is a good intro for reminding them why they need to learn to work with JSON, but let's also be specific about the type of work theyre doing in this lab

File: README.md [hunk 2]
```diff
@@ -1 +1,43 @@
...
+## Instructions
+
+1. Create a file `database.json` in `config` folder
+2. Copy object from the snippet below into `module.js` and convert it to a **valid** JSON
+3. Create a file `main.js` 
+4. Import your configuration module in `main.js` with `require()`
+5. Use `path.join()` and `__dirname` for the path to `database.json`
+5. In `main.js`, extract the `domain` and `port` values of the `dev` property, and print them with `console.log()` separated by `:`.
+6. Keep `main.js` in the repository root folder (folder of this `README.md` file)
+6. Run `$ npm test` to verify
```
Comment (victhevenot) [inline README.md]: remove `$`

File: README.md [hunk 3]
```diff
@@ -1 +1,43 @@
...
+
+In this lab, you'll learn how to import JSON files. This can be useful for data (e.g., list of customers) or configuration (e.g., server settings). In other words, JSON files are good for storing static information, i.e., information without JavaScript/Node code.
+
+## Instructions
+
+1. Create a file `database.json` in `config` folder
+2. Copy object from the snippet below into `module.js` and convert it to a **valid** JSON
+3. Create a file `main.js` 
+4. Import your configuration module in `main.js` with `require()`
+5. Use `path.join()` and `__dirname` for the path to `database.json`
```
Comment (victhevenot) [inline README.md]: what is `_dirname`? previous reading didn't cover it.

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+var test = require('tape'),
```
Comment (victhevenot) [inline test/index.js]: test still uses Tape - and refactor for readability

**Comment by azat-co** [comment]
updated CC @AnnJohn @vicfriedman

**Comment by victhevenot** [comment]
going to @pletcher next, but a few comments for you @azat-co. And make sure wip-master gets these changes!
- The instructions in the `README.md` that have students copy and object from the readme and place it in file is just an extra step that doesn't teach the student anything. lets just give them the file with the object and change the instructions there to just convert it to valid JSON
- Line 9 of the tests are effectively giving the students the answer for the code to write in `main.js`. I think it would be better to hardcode in the exact string we expect `main` to return.
- The second test (lines 23 and 24) pass if `_dirname` and `.join()` are in `main.js` as strings

File: README.md [hunk 5]
```diff
@@ -1 +1,45 @@
-# node-modules-json-lab
\ No newline at end of file
+# Importing JSON Files Lab
+
+## Objectives
+
+1. Implement a JSON config file
```
Comment (pletcher) [inline README.md:5]: I'm not sure "implement" is the right word here. Really, the third objective sums it all up -- "Use an external JSON configuration file in a script"

**Comment by pletcher** [comment]
I added one quick comment in addition to @vicfriedman, but my main concern with this lab is that it doesn't ask students to do much. Could we have them build something a bit more substantial?

Granted that they haven't experienced much at this point, but (this is just the first idea that popped into my head and it may well be a terrible way) they have done work on `fs`, modules, timeouts, and configuration -- why not have them build a rudimentary file watcher? It could poll for changes in a given file (both the polling interval and the file to watch could be set in the config), and just log out things like "The file has not changed since ${time}" or "The file just changed!"

Maybe that's not the best example, but just something that gives them a bit more to bite into :)

**Pull request closed without merging**